samples/quickstart-guice/src/main/java/QuickstartShiroModule.java

```diff
@@ -16,22 +16,94 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import com.google.inject.Key;
+import com.google.inject.Provider;
 import com.google.inject.Provides;
+import com.google.inject.TypeLiteral;
+import com.google.inject.matcher.Matchers;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.name.Names;
+import com.google.inject.spi.InjectionListener;
+import com.google.inject.spi.TypeEncounter;
+import com.google.inject.spi.TypeListener;
 import org.apache.shiro.config.Ini;
+import org.apache.shiro.config.event.LoggingBeanEventListener;
+import org.apache.shiro.event.Event;
+import org.apache.shiro.event.EventBus;
+import org.apache.shiro.event.Subscribe;
+import org.apache.shiro.event.support.AnnotationEventListenerResolver;
+import org.apache.shiro.event.support.DefaultEventBus;
 import org.apache.shiro.guice.ShiroModule;
 import org.apache.shiro.realm.text.IniRealm;
+import org.apache.shiro.util.ClassUtils;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Set;
 
 public class QuickstartShiroModule extends ShiroModule {
+
+    private EventBus eventBus;
+
+    @Provides
+    @Singleton
+    public EventBus getEventBus(@Named("eventListeners") Set<Object> eventListeners) {
+        // ignore the eventListeners, we just want to force the initial loading of these.
+        if (eventBus == null) {
+            eventBus = new DefaultEventBus();
+        }
+        return eventBus;
+    }
+
+    @Provides
+    Ini loadShiroIni() {
+        return Ini.fromResourcePath("classpath:shiro.ini");
+    }
+
     protected void configureShiro() {
+
+        bindConstant().annotatedWith(Names.named("shiro.securityManager.eventBus")).to(EventBus.class);
+        expose(EventBus.class);
+
+        configureEventListeners(Multibinder.newSetBinder(binder(), Object.class, Names.named("eventListeners")));
+
+
+        bind(Key.get(LoggingBeanEventListener.class));
```

Comment:
this listener is referenced twice, once here, and once in the `configureEventListeners` method.